curation-migration-datasync/index.ts

```diff
@@ -1,22 +1,57 @@
 import config from './config';
 import * as Sentry from '@sentry/serverless';
 import { writeClient } from './database/dbClient';
-import { EventBridgeEvent } from 'aws-lambda';
-import { addScheduledItem } from './eventConsumer';
-import { EventDetailType } from './types';
+import {
+  SQSEvent,
+  SQSBatchResponse,
+  SQSBatchItemFailure,
+  EventBridgeEvent,
+} from 'aws-lambda';
+import { addScheduledItem, removeScheduledItem } from './eventConsumer';
+import { EventDetailType, ScheduledItemPayload } from './types';
 
-export async function handlerFn(event: EventBridgeEvent<any, any>) {
-  console.log(JSON.stringify(event));
+/**
+ * Handler entrypoint. Loops over every record in the message and calls
+ * the function logic to handle the record. Reports the batch failures
+ * (partial success is possible).
+ * @param event the SQS message
+ * @returns the batch failures, if any
+ */
+export async function handlerFn(event: SQSEvent): Promise<SQSBatchResponse> {
```

Comment:
This is to separate the looping/failure reporting behavior from the actual message handling (`_handlerFn`)